### ISSUE TITLE
Bugs...

### ISSUE THREAD

**Issue by Caesar-s1mple** [ansonb/RECON#13]
Hello, after serveral days of debugging, I have found many bugs in your code.
I'd like to share them with other people, and I hope you can fix them.
(1)Tons of file path errors which I don't want to list here.
(2)models/models.py
![image](https://user-images.githubusercontent.com/73978771/169635629-e68e26ab-87b6-4ee6-92c2-cb246d32fb6f.png)
Why you use nn.Parameter to create 'init_gat_relation_embeddings' here? Moreover, why you set 'requires_grad=True'?
(3)GAT_seq_space/main.py
- set '--get_1hop' and '--get_2hop' to True on your first run.
- 
![image](https://user-images.githubusercontent.com/73978771/169635833-01937a4a-e385-4300-897b-d4fb410f2341.png)
'outfolder' is not defined...^^

- 
![image](https://user-images.githubusercontent.com/73978771/169635903-489dd526-7555-40ea-8435-40d15a664278.png)
I don't know why you write these.
The proper code is
```
model_gat.load_state_dict(torch.load(
        os.path.join(args.output_folder, 'trained_{}.pth'.format(args.epochs_gat - 1))))
```
the same bug also exists on many other lines...

(4)train.py
Too many...I just wanna list some of them
- lines 54 - 66
Code here is not complete, resulting a lot of 'not defined' errors.
- A lot of inappropriate using, like
```
torch.from_numpy(***.astype(int))
```
- Did you really train this model with ONLY 10 epochs? Seriously?
- File paths written in fron are all about wikidata, and then you load data using data='nyt'... ^^
I know it's not a big deal, but I seems to prove that your code is ..... :(
(5)Tons of other bugs that I have fixed and forgotten

Thanks a lot if you can fix them all :)
And it will be more thankful if you can write CLEARLY how to run your code in README.md 👍

**Comment by nadgeri14** [comment]
Hi, 

Thanks for pointing out the issues. We will look into the issues and fix them and make sure that it does not happen again in future :D 

Some immediate clarifications I would like to make are the following - 

- In the file GAT_seq_space/main.py you suggested the code for loading model `model_gat.load_state_dict` but if you look at the commented line 788 it is exactly the same that you suggested :) 
- And yes, we did train the model for 10 epochs only :)

**Comment by Caesar-s1mple** [comment]
Thanks for your reply. 👍

Another quesion, I can't figure out why I can't run train.py for over one time.
For example, I trained GAT in GAT_seq_space, and got 'final_entity_embeddings.json', 'final_relation_embeddings.json' and 'W_ent2rel.json.npy'.
Then I used main.py to train RECON. 
During the training, everything worked well, f1-score on validation set went from 0 to 0.65 after 10 epochs on my own dataset.
But when I want to train RECON for the twice time(I didn't change anything), it seemed that the training process is terribly wrong, cuz f1-score on validation set doesn't go up at all(remaining below 0.002).
Then I deleted models/RECON and restarted train.py to train RECON, but the same problem still existed.
All I can do is to re-train GAT in GAT_seq_space and get 'final_entity_embeddings.json', 'final_relation_embeddings.json' and 'W_ent2rel.json.npy' one more time. Then I can get expected f1-score when using train.py for the FIRST time.
This is really confusing... I haven't found out where the bug is yet...

**Comment by VinsonZhangS** [comment]
他们模型的bug真的很多，他们的代码和thunlp的GPGNN是很相似的，如果你在训练中遇到了indices没有定义的问题，可以看看GPGNN的test.py

**Comment by Caesar-s1mple** [comment]
> 他们模型的bug真的很多，他们的代码和thunlp的GPGNN是很相似的，如果你在训练中遇到了indices没有定义的问题，可以看看GPGNN的test.py

感谢，不过经过我几天的修改，他们代码的所有bug基本都被我修复了，甚至还加了本来应该有的很多功能及评估指标。
希望这个模型能给点力吧，不枉我修复这么久。

**Comment by VinsonZhangS** [comment]
> > 他们模型的bug真的很多，他们的代码和thunlp的GPGNN是很相似的，如果你在训练中遇到了indices没有定义的问题，可以看看GPGNN的test.py
> 
> 感谢，不过经过我几天的修改，他们代码的所有bug基本都被我修复了，甚至还加了本来应该有的很多功能及评估指标。 希望这个模型能给点力吧，不枉我修复这么久。

同学你好，请问可不可以分享一下你的相关代码？如果不方便的话可不可以细说一下添加的相关功能和评估指标

**Comment by Caesar-s1mple** [comment]
> > > 他们模型的bug真的很多，他们的代码和thunlp的GPGNN是很相似的，如果你在训练中遇到了indices没有定义的问题，可以看看GPGNN的test.py
> > 
> > 
> > 感谢，不过经过我几天的修改，他们代码的所有bug基本都被我修复了，甚至还加了本来应该有的很多功能及评估指标。 希望这个模型能给点力吧，不枉我修复这么久。
> 
> 同学你好，请问可不可以分享一下你的相关代码？如果不方便的话可不可以细说一下添加的相关功能和评估指标

当然可以，这两天我会把相关代码分享出来。

**Comment by giusidp** [comment]
> 

Hi Caesar, I've translated your comment and if I'm right you said you would have uploaded the fixed code in two days? It would really help me a bug fixed version of this code, if you want you can contact me to the e-mail shown on my profile!  ^^

**Comment by giusidp** [comment]
> > > > 他们模型的bug真的很多，他们的代码和thunlp的GPGNN是很相似的，如果你在训练中遇到了indices没有定义的问题，可以看看GPGNN的test.py
> > > 
> > > 
> > > 感谢，不过经过我几天的修改，他们代码的所有bug基本都被我修复了，甚至还加了本来应该有的很多功能及评估指标。 希望这个模型能给点力吧，不枉我修复这么久。
> > 
> > 
> > 同学你好，请问可不可以分享一下你的相关代码？如果不方便的话可不可以细说一下添加的相关功能和评估指标
> 
> 当然可以，这两天我会把相关代码分享出来。



> > > > 他们模型的bug真的很多，他们的代码和thunlp的GPGNN是很相似的，如果你在训练中遇到了indices没有定义的问题，可以看看GPGNN的test.py
> > > 
> > > 
> > > 感谢，不过经过我几天的修改，他们代码的所有bug基本都被我修复了，甚至还加了本来应该有的很多功能及评估指标。 希望这个模型能给点力吧，不枉我修复这么久。
> > 
> > 
> > 同学你好，请问可不可以分享一下你的相关代码？如果不方便的话可不可以细说一下添加的相关功能和评估指标
> 
> 当然可以，这两天我会把相关代码分享出来。

Hi Caesar, I've translated your comment and if I'm right you said you would have uploaded the fixed code in two days? It would really help me a bug fixed version of this code, if you want you can contact me to the e-mail shown on my profile! ^^

**Comment by giusidp** [comment]
> > > > 他们模型的bug真的很多，他们的代码和thunlp的GPGNN是很相似的，如果你在训练中遇到了indices没有定义的问题，可以看看GPGNN的test.py
> > > 
> > > 
> > > 感谢，不过经过我几天的修改，他们代码的所有bug基本都被我修复了，甚至还加了本来应该有的很多功能及评估指标。 希望这个模型能给点力吧，不枉我修复这么久。
> > 
> > 
> > 同学你好，请问可不可以分享一下你的相关代码？如果不方便的话可不可以细说一下添加的相关功能和评估指标
> 
> 当然可以，这两天我会把相关代码分享出来。

Hi @Caesar-s1mple  I've translated your comment and if I'm right you said you would have uploaded the fixed code in two days? It would really help me a bug fixed version of this code, if you want you can contact me to the e-mail shown on my profile! ^^

**Comment by LOUGBERTI** [comment]
> > 他们模型的bug真的很多，他们的代码和thunlp的GPGNN是很相似的，如果你在训练中遇到了indices没有定义的问题，可以看看GPGNN的test.py
> 
> 感谢，不过经过我几天的修改，他们代码的所有bug基本都被我修复了，甚至还加了本来应该有的很多功能及评估指标。 希望这个模型能给点力吧，不枉我修复这么久。

您好，请问您可以放一下您的代码嘛？我被这个代码困扰很久了

**Comment by LOUGBERTI** [comment]
> > > > 他们模型的bug真的很多，他们的代码和thunlp的GPGNN是很相似的，如果你在训练中遇到了indices没有定义的问题，可以看看GPGNN的test.py
> > > 
> > > 
> > > 感谢，不过经过我几天的修改，他们代码的所有bug基本都被我修复了，甚至还加了本来应该有的很多功能及评估指标。 希望这个模型能给点力吧，不枉我修复这么久。
> > 
> > 
> > 同学你好，请问可不可以分享一下你的相关代码？如果不方便的话可不可以细说一下添加的相关功能和评估指标
> 
> 当然可以，这两天我会把相关代码分享出来。

同学您好，可以公布一下您的代码嘛？我弄了一个月了都没有跑通，可不可以联系一下我。1453646213@qq.com
